pkgs/development/go-modules/generic/default.nix

```diff
@@ -198,7 +198,7 @@ go.stdenv.mkDerivation (
      ln -s "${dep.src}" "$d/src/${dep.goPackagePath}"
   ''
   ) goPath) + ''
-    export GOPATH="$d:$GOPATH"
+    export GOPATH=${lib.concatStringsSep ":" ( ["$d"] ++ ["$GOPATH"] ++ ["$PWD"] ++ extraSrcPaths)}
```

Comment:
@qknight After thinking about it, it is a sane default. If someone would have a GOPATH set, it would still do things as before otherwise it write to the nix-shell invocation directory.